src/librustc/ty/mod.rs

```diff
@@ -1694,7 +1694,7 @@ impl<'a, 'gcx, 'tcx> AdtDef {
     /// Asserts this is a struct and returns the struct's unique
     /// variant.
     pub fn struct_variant(&self) -> &VariantDef {
-        assert!(!self.is_enum());
+        assert!(self.is_struct());
```

Comment:
Things could change, but IIRC it was called on unions and it was *supposed* to be called on unions, because unions are represented as an ADT with a single "variant".

Reply:
In that case the name is misleading and the doc comment is wrong. I'm now auditing all callers to see whether they might be using this on unions. Depending on results I might change the doc comment and possibly the name.

Reply:
`non_enum_variant` is what I'd use.

Reply:
Alternatively, introduce another method that does the same thing for unions (if union code wants the helper method but there's no code that needs to work on structs *and* unions).

Reply:
I'd expect most code using this is common for structs and unions.

Reply:
Yeah there's a bunch. `non_enum_variant` it is.